KILTprotocol/mashnet-node
# pay the author

## fixes KILTprotocol/ticket/issues/664

what did change?

We are know using the session pallet! The authority keys can therefore change for each session (a specific period of block numbers). I guess we don't want to worry about the sessions know, so i set the session length to 0xFFFF_FFFF_FFFF_FFFF blocks. The sessions return specific authoring keys for an account. We therefore have access to a mapping from AccountID to Authoring Keys. Without the session pallet the block author has no identity.

## How to test:

make a tranksaction and verify that the `total_issuance` didn't change and the block author received the fee.

* check that the total issuance doesn't change after making transactions (`console.log(await api.query.balances.totalIssuance())`)
* check that the block author receives funds although he didn't get any transfers (`console.log(await api.query.system.account("5FA9nQDVg267DEd8m1ZypXLBnvN7SFxYwV7ndqSYGiN9TTpu"))` before and after a transaction)


## Checklist:
